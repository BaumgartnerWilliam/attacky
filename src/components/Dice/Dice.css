@keyframes spinIt {
    from {
        transform: rotate(0);
    }
    to {
        transform: rotate(360deg);
    }
}

.dice-animating {
    animation-name: spinIt;
    animation-duration: 1.5s;
    animation-iteration-count: infinite;
}
