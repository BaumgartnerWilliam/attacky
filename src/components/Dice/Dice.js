import React from 'react';
import { Row } from 'react-bootstrap';
import './Dice.css';

import dice0 from '../../assets/dice0.png';
import dice1 from '../../assets/dice1.png';
import dice2 from '../../assets/dice2.png';
import dice3 from '../../assets/dice3.png';
import dice4 from '../../assets/dice4.png';
import dice5 from '../../assets/dice5.png';
import dice6 from '../../assets/dice6.png';

const diceAssets = {
  0: dice0,
  1: dice1,
  2: dice2,
  3: dice3,
  4: dice4,
  5: dice5,
  6: dice6
};

// 
const Dice = ({ value, animate }) => (
  <Row data-testid="dice" className="justify-content-center pt-4">
    {/* we can use classNames lib as well */}
    <img
      className={`${animate && 'dice-animating'}`}
      data-value={value}
      data-testid="value"
      src={animate && diceAssets[0] || diceAssets[value]}
      style={{ width: '70px' }}
    />
  </Row>
);

export default Dice;
