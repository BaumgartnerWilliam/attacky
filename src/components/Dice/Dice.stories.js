import React from 'react';
import { withKnobs, number } from '@storybook/addon-knobs';

import Dice from './Dice';


export default {
  title: 'Dice',
  decorators: [withKnobs]
};


export const withValue = () => <Dice value={number('value', 3)} />
