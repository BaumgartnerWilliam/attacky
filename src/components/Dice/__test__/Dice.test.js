import React from 'react';
import { render } from '@testing-library/react';
import Dice from '../Dice';

describe('dice component', () => {
  it('renders without crashing', () => {
    const { container } = render(<Dice />);
    expect(container).toBeInTheDocument();
  });

  it('sets data-value properly', () => {
    const { getByTestId } = render(<Dice value={"5"} />);
    const el = getByTestId('value');
    expect(el).toHaveAttribute('data-value', "5");
  });

  it('matches snapshot', () => {
    const { getByTestId } = render(<Dice value={"5"} />);
    const testid = getByTestId('dice');
    expect(testid).toMatchSnapshot();
  });
});
