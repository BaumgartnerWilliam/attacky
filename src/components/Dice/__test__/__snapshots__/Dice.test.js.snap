// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dice component matches snapshot 1`] = `
<div
  class="justify-content-center pt-4 row"
  data-testid="dice"
>
  <img
    class="undefined"
    data-testid="value"
    data-value="5"
    src="dice5.png"
    style="width: 70px;"
  />
</div>
`;
